feat: LEAP-1119: Feature/gridview thumbnail

FrsECM opened the pull request:
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend

### Describe the reason for change
When using multilayered images, the original behaviour was to create a src that separate each image by a ','.

This way, it's not displayable.
![image](https://github.com/HumanSignal/label-studio/assets/26071804/cf8fd01b-fc47-4ada-a97d-fc8ff405a647)

#### What does this fix?
It fixes an issue in the gridview when there is multiple layers image.
![image](https://github.com/HumanSignal/label-studio/assets/26071804/ee02ba17-f716-4e8a-870c-5721d4d9256d)
It's even possible to use a specific "thumbnail" field, that allows to add a specific thumbnail to be use.
![image](https://github.com/HumanSignal/label-studio/assets/26071804/b849f0bd-8532-44d0-82e5-09737cd5a66c)
If you want the thumbnail to be displayed in the column view, just declare it as an image in the taxonomy : 
![image](https://github.com/HumanSignal/label-studio/assets/26071804/f4f862ac-1fee-4a9b-a84d-ea3f03ed9ef3)
![image](https://github.com/HumanSignal/label-studio/assets/26071804/af0287da-e15d-4f7a-9855-02c093cf76a4)


#### What is the new behavior?
There is two different new behaviors.
If a "thumbnail" is provided : 
- The gridview uses the thumbnail field.
The advantage is the ability to use a thumbnail that has a lower resolution than the original image.
else:
We use the first image of the multilayer images.

#### What is the current behavior?
The current behavior is that in this case, images url are separated by a ',' in the image src.
What display nothing at all

#### Does this change affect performance?
It can improve the performance because thumbnail can be a lower resolution image.


### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit

### Which logical domain(s) does this change affect?
GridView => Data Exploration

FrsECM committed:
Add Thumbnail + Fix bug with Multilayers

FrsECM committed:
Add etc to gitignore

sajarin commented:
/jira create
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/9289869308)
> Jira issue [TRIAG-577](https://humansignal.atlassian.net/browse/TRIAG-577) is created

[TRIAG-577]: https://humansignal.atlassian.net/browse/TRIAG-577?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

juliosgarbi committed:
merge with develop

juliosgarbi committed:
fix the file extension

juliosgarbi committed:
fix biome

juliosgarbi committed:
fix biome check

juliosgarbi committed:
fix unit tests for datamanager and add tests for the change

juliosgarbi committed:
remove the validation for thumbnail in the grid view, since the naming is not consistent and it can create confusion

juliosgarbi committed:
fix biomejs

juliosgarbi committed:
fix biome checks

juliosgarbi committed:
fix biome checks

juliosgarbi committed:
fix unit tests

juliosgarbi reviewed (approved):
I added to the PR unit tests, also I removed the feature about show thumbnails, we already have something similar to it on the list view that doesn't need add any tags on the config, you just need to add a new node on the data (e.g. thumb), enable this column on the columns dropdown, and change the view mode from `str` to `img` on the thumb column, overall it looks good, thanks for the contribution!

FrsECM commented:
> I added to the PR unit tests, also I removed the feature about show thumbnails, we already have something similar to it on the list view that doesn't need add any tags on the config, you just need to add a new node on the data (e.g. thumb), enable this column on the columns dropdown, and change the view mode from `str` to `img` on the thumb column, overall it looks good, thanks for the contribution!

If you have a piece of document about that I'm very interessed. It can help when we have very high resolution images. 

Thanks !
